adjust.go

```diff
@@ -154,73 +157,178 @@ func (f *File) adjustRowDimensions(sheet string, ws *xlsxWorksheet, row, offset
+	if topCol >= num {
+		topCol += offset
+	}
+	if bottomCol >= num {
+		bottomCol += offset
+	}
+
+	topColName, _ := ColumnNumberToName(topCol)
+	bottomColName, _ := ColumnNumberToName(bottomCol)
+	return fmt.Sprintf("%s:%s", topColName, bottomColName)
+}
+
+// adjustFormula provides a function to adjust formula reference and shared
+// formula reference.
+func (f *File) adjustFormula(sheet string, formula *xlsxF, dir adjustDirection, num, offset int, si bool) error {
+	if formula == nil {
+		return nil
+	}
+
 	var err error
 	if formula.Ref != "" {
-		if formula.Ref, err = adjustRef(formula.Ref); err != nil {
+		if formula.Ref, err = f.adjustRef(formula.Ref, dir, num, offset); err != nil {
 			return err
 		}
 		if si && formula.Si != nil {
 			formula.Si = intPtr(*formula.Si + 1)
 		}
 	}
 	if formula.T == STCellFormulaTypeArray {
-		formula.Content, err = adjustRef(strings.TrimPrefix(formula.Content, "="))
+		formula.Content, err = f.adjustRef(strings.TrimPrefix(formula.Content, "="), dir, num, offset)
 		return err
 	}
-	if formula.Content != "" && !strings.ContainsAny(formula.Content, "[:]") {
+	if formula.Content != "" && !strings.ContainsAny(formula.Content, "[:]()") {
 		content, err := f.adjustFormulaRef(sheet, formula.Content, dir, num, offset)
 		if err != nil {
 			return err
 		}
 		formula.Content = content
+	} else if formula.Content != "" {
+		ps := efp.ExcelParser()
+		content := ""
+		for _, token := range ps.Parse(formula.Content) {
+			if f.isRangeOperand(token) {
+				// If string contains ":" and both sides have at least one letter and one number,
+				// create a regex for this expression
+				content = f.handleRangeOperand(sheet, token, dir, num, offset, content)
+			} else if f.isFunctionStart(token) {
+				content += token.TValue + "("
+			} else if f.isFunctionStop(token) {
+				content += token.TValue + ")"
+			} else {
+				content += token.TValue
+			}
+		}
+		formula.Content = content
 	}
 	return nil
 }
 
+// isFunctionStop provides a function to check if token is a function stop.
+func (f *File) isFunctionStop(token efp.Token) bool {
+	return token.TType == efp.TokenTypeFunction && token.TSubType == efp.TokenSubTypeStop
+}
+
+// isFunctionStart provides a function to check if token is a function start.
+func (f *File) isFunctionStart(token efp.Token) bool {
+	return token.TType == efp.TokenTypeFunction && token.TSubType == efp.TokenSubTypeStart
+}
+
+// isRangeOperand provides a function to check if token is a range operand.
+func (f *File) isRangeOperand(token efp.Token) bool {
+	return token.TType == efp.TokenTypeOperand && token.TSubType == efp.TokenSubTypeRange
+}
+
+// handleRangeOperand provides a function to handle range operand tokens.
+// Examples are A2:C5, 1:2, A:B
+func (f *File) handleRangeOperand(sheet string, token efp.Token, dir adjustDirection, num int, offset int, content string) string {
+	if regex := regexp.MustCompile("[A-Z]+[0-9]+:[A-Z]+[0-9]+"); regex.MatchString(token.TValue) {
```

Comment:
Can we using the rune range instead of regular expression to detect the operand, that would be better for performance.